Formula/perl.rb

```diff
@@ -44,7 +46,16 @@ def install
       -Duselargefiles
       -Dusethreads
     ]
-    args << "-Dsed=/usr/bin/sed" if OS.mac?
+
+    if OS.mac?
+      args << "-Dsed=/usr/bin/sed"
+    else
+      # Override libswanted so it doesn't try to link to nsl, which is deprecated from glibc.
+      args += [
+        "-A",
+        "define:libswanted=\"cl pthread socket bind gdbm db inet malloc dl ld m crypt sec util c cposix posix\"",
```

Comment:
I'm a bit wary of hardcoding a list like this here. It seems really prone to bitrot.

Reply:
I'm going to drop the `libnsl` fix for now to focus on the `libxcrypt` one.  That is a critical breakage affecting all users with `glibc` installed, while the `libnsl` one only affects building from source.